File: problem/Digits_Equal_I/hasSameDigits.java
class Solution {
    public boolean hasSameDigits(String s) {
        char[] arr = s.toCharArray();
        int iteration = 0;
        
        while (arr.length - iteration != 2) {
            for (int i = 0; i < arr.length - 1 - iteration; i++) {
                arr[i] = (char)(((arr[i] - '0') + (arr[i + 1] - '0')) % 10 + '0');
            }
            iteration++;
        }
        
        return arr[0] == arr[1];
    }
}

File: problem/Digits_Equal_I/hasSameDigits.py
class Solution:
    def hasSameDigits(self, s):
        s = list(s)
        iteration = 0
        
        while len(s) - iteration != 2:
            for i in range(len(s) - 1 - iteration):
                s[i] = str((int(s[i]) + int(s[i + 1])) % 10)
            iteration += 1
        
        return s[0] == s[1]